app/javascript/alchemy_admin/components/datepicker.js

```diff
@@ -23,7 +28,16 @@ class Datepicker extends AlchemyHTMLElement {
       }
     }
 
-    flatpickr(this.getElementsByTagName("input")[0], options)
+    // if flatpickr was initialized earlier, destroy it first and then initialize it again
+    if (this.flatpickr) {
+      this.flatpickr.destroy()
+    }
```

Comment:
Why is this bandaid fix necessary although we destroy flatpickr in the disconnected callback? Is there something weird happening in our render code?

Reply:
Great find!